rdwatch/core/tasks/animation_export.py

```diff
@@ -0,0 +1,802 @@
+
+from django.contrib.auth.models import User
+from django.core.files import File
+from django.db.models import Q
+
+from rdwatch.celery import app
+from rdwatch.core.models import (
+    AnimationModelRunExport,
+    AnimationSiteExport,
+    ModelRun,
+    SiteEvaluation,
+    SiteImage,
+)
+
+logger = logging.getLogger(__name__)
+
+label_mapping = {
+    'active_construction': {
+        'color': (192, 0, 0),
+        'label': 'Active Construction',
+    },
+    'post_construction': {
+        'color': (91, 155, 213),
+        'label': 'Post Construction',
+    },
+    'site_preparation': {
+        'color': (255, 217, 102),
+        'label': 'Site Preparation',
+    },
+    'unknown': {
+        'color': (112, 48, 160),
+        'label': 'Unknown',
+    },
+    'no_activity': {
+        'color': (166, 166, 166),
+        'label': 'No Activity',
+    },
+    'positive_annotated': {
+        'color': (127, 0, 0),
+        'label': 'Positive Annotated',
+    },
+    'positive': {
+        'color': (127, 0, 0),
+        'label': 'Positive Annotated',
+    },
+    'positive_partial': {
+        'color': (0, 0, 139),
+        'label': 'Positive Partial',
+    },
+    'positive_annotated_static': {
+        'color': (255, 140, 0),
+        'label': 'Positive Annotated Static',
+    },
+    'positive_partial_static': {
+        'color': (255, 255, 0),
+        'label': 'Positive Partial Static',
+    },
+    'positive_pending': {
+        'color': (30, 144, 255),
+        'label': 'Positive Pending',
+    },
+    'positive_excluded': {
+        'color': (0, 255, 255),
+        'label': 'Positive Excluded',
+    },
+    'negative': {
+        'color': (255, 0, 255),
+        'label': 'Negative',
+    },
+    'ignore': {
+        'color': (0, 255, 0),
+        'label': 'Ignore',
+    },
+    'transient_positive': {
+        'color': (255, 105, 180),
+        'label': 'Transient Positive',
+    },
+    'transient_negative': {
+        'color': (255, 228, 196),
+        'label': 'Transient Negative',
+    },
+    'system_proposed': {
+        'color': (31, 119, 180),
+        'label': 'System Proposed',
+    },
+    'system_confirmed': {
+        'color': (31, 119, 180),
+        'label': 'System Confirmed',
+    },
+    'system_rejected': {
+        'color': (31, 119, 180),
+        'label': 'System Rejected',
+    },
+}
+
+
+def rescale_bbox(bbox, scale):
+    """
+    Rescales a bounding box by a given scaling factor.
+
+    Parameters:
+    bbox (list): A list containing [minx, miny, maxx, maxy].
+    scale (float): The scaling factor to apply to the bounding box.
+
+    Returns:
+    list: A list containing the rescaled [minx, miny, maxx, maxy].
+    """
+    minx, miny, maxx, maxy = bbox
+
+    center_x = (minx + maxx) / 2
+    center_y = (miny + maxy) / 2
+
+    width = (maxx - minx) * scale / 2
+    height = (maxy - miny) * scale / 2
+
+    return [center_x - width, center_y - height, center_x + width, center_y + height]
```

Comment:
The `.extent` property from a geodjango polygon is actually a `tuple`, not a `list`. I would also suggest changing the return type to be consistent with that.

```suggestion
def rescale_bbox(bbox: tuple[float, float, float, float], scale: float) -> tuple[float, float, float, float]:
    """
    Rescales a bounding box by a given scaling factor.

    Parameters:
    bbox (tuple): A tuple containing [minx, miny, maxx, maxy].
    scale (float): The scaling factor to apply to the bounding box.

    Returns:
    list: A list containing the rescaled [minx, miny, maxx, maxy].
    """
    minx, miny, maxx, maxy = bbox

    center_x = (minx + maxx) / 2
    center_y = (miny + maxy) / 2

    width = (maxx - minx) * scale / 2
    height = (maxy - miny) * scale / 2

    return (center_x - width, center_y - height, center_x + width, center_y + height)
```